pkg/stargate/stargate.go

```diff
@@ -248,18 +32,14 @@ func NewService(sg *api.Stargate, dc *cassdcapi.CassandraDatacenter) *corev1.Ser
 				{Port: 9042, Name: "cassandra"},
 			},
 			Selector: map[string]string{
-				api.StargateLabel: serviceName,
+				api.StargateLabel: stargate.Name,
```

Comment:
I believe this was wrong since the beginning. The selector should match all pods with `StargateLabel` = `stargate.Name`, not `serviceName`.

Reply:
Confirmed that this is now working as expected.